rama-tcp/src/client/connect.rs

```diff
@@ -167,37 +178,32 @@ where
     let connected = Arc::new(AtomicBool::new(false));
     let sem = Arc::new(Semaphore::new(3));
 
-    // IPv6
-    let ipv6_tx = tx.clone();
-    let ipv6_domain = domain.clone();
-    let ipv6_connected = connected.clone();
-    let ipv6_sem = sem.clone();
-    ctx.spawn(tcp_connect_inner_branch(
-        dns.clone(),
-        connector.clone(),
-        IpKind::Ipv6,
-        ipv6_domain,
-        port,
-        ipv6_tx,
-        ipv6_connected,
-        ipv6_sem,
-    ));
+    if dns.ipv4_supported() {
+            ctx.spawn(tcp_connect_inner_branch(
+                dns.clone(),
```

Comment:
You can add `Copy` derive trait to these mode Enums, then you don't need to Clone